### PR TITLE
Refactor Git proxy

### PR DESCRIPTION
**Pull request by janfjohannes**
this branch has a preview deployment at: [git.inlang.com](https://git.inlang.com/)

can be tested locally by switching the editor to use the external git proxy:

 PUBLIC_GIT_PROXY_BASE_URL=http://localhost:3001 doppler run --preserve-env -- npm run dev
 

 
 check that the editor only accesses port 3001 for git and github actions
 
 double check login, logout, open repo + push work as expected
 
 
 merging this PR will not enable the external git proxy until the environment variable is changed on doppler/render.com

### CODE REVIEW

**Comment by samuelstroschein** [comment]
@janfjohannes outdated?

**Comment by janfjohannes** [comment]
@samuelstroschein  yes i will push the external git proxy server here today. this PR  is just empty as it was safer to first merge the cors related changes that we did yesterday

**Comment by samuelstroschein** [comment]
@janfjohannes should i review this pr or not? 

I am confused since this PR seems work in progress as indicated by frequent commits

**Comment by janfjohannes** [comment]
@samuelstroschein  yes please, you mean i should sqash commits before asking for a review or just a mtter of caution to see if all commits were pushed before you look at it?

**Comment by samuelstroschein** [comment]
@janfjohannes unrelated to squashing commits. i would expect a review if something is done or a comment that i should review now _but this is still work in progress_. neither happened which made me wonder if you clicked on review by accident

File: lix/source-code/server/tsconfig.json [hunk 1]
```diff
@@ -2,8 +2,12 @@
 	"extends": "../tsconfig.base.json",
 	"include": ["src/**/*"],
 	"compilerOptions": {
-		"lib": ["ESNext", "DOM"],
-		"types": [],
+		"lib": ["DOM", "DOM.Iterable", "ESNext"],
+		"types": ["vite/client"],
```
Comment (samuelstroschein) [inline lix/source-code/server/tsconfig.json:6]: Why is are vite types required?

File: lix/source-code/server/tsconfig.json [hunk 2]
```diff
@@ -2,8 +2,12 @@
...
+		"jsx": "preserve",
+		"jsxImportSource": "solid-js",
+		"esModuleInterop": true,
+		"resolveJsonModule": true,
```
Comment (samuelstroschein) [inline lix/source-code/server/tsconfig.json:10]: Copy & paste error?

File: lix/source-code/server/package.json [hunk 3]
```diff
@@ -11,17 +11,34 @@
...
+		"@isomorphic-git/cors-proxy": "2.7.1",
+		"@sentry/node": "^7.47.0",
+		"@sentry/tracing": "^7.47.0",
+		"compression": "^1.7.4",
+		"cookie-session": "2.0.0",
+		"express": "^4.18.2",
+		"jose": "4.14.6",
+		"octokit": "3.1.0",
+		"tsx": "3.12.8",
+		"vite": "4.4.9"
```
Comment (samuelstroschein) [inline lix/source-code/server/package.json:33]: copy & paste error in the dependencies too? vite as a dependency seems wrong

**Review by samuelstroschein — APPROVED** [review]
Seems like nothing changed. The dependencies seem wrong though.

**Pull request merged**